covid_graphs/covid_graphs/scripts/rest.py

```diff
@@ -0,0 +1,105 @@
+from pathlib import Path
+from typing import Dict, List
+
+from flask import Flask, abort, jsonify
+
+from covid_graphs.country_graph import CountryGraph
+from covid_graphs.country_report import create_report
+from covid_graphs.predictions import prediction_db
+
+TITLE = "COVID-19 predictions of Boďová and Kollár"
+CountryGraphsByReportName = Dict[str, List[CountryGraph]]
+
+
+class Rest:
+    def __init__(self, data_dir: Path):
+        # vytvor strukturu pre predikcie a krajiny
+        # vytvor strukturu pre data
+        self.available_predictions = Rest._create_available_predictions_by_date()
+        self.country_reports = Rest._create_country_reports(data_dir)
+        predictions, country_reports_active = Rest._create_predictions(self.country_reports)
+        self.predictions = predictions
+        self.country_reports_active = country_reports_active
+
+    @staticmethod
+    def _create_available_predictions_by_date():
+        result = {}
+        for x in prediction_db.get_prediction_events():
+            result[x.name] = {
+                "label": x.date,
+                "countries": [p.country for p in prediction_db.predictions_for_event(x)],
+            }
+        return result
+
+    @staticmethod
+    def _create_country_reports(data_dir: Path):
+        country_reports = {}
+        for country_short_name in prediction_db.get_countries():
+            if (data_dir / f"{country_short_name}.data").is_file():
+                country_report = create_report(
+                    data_dir / f"{country_short_name}.data", country_short_name
+                )
+                country_reports[country_report.short_name] = country_report
+
+        return country_reports
+
+    @staticmethod
+    def _create_predictions(country_reports):
```

Comment:
Although we don't enforce types, the typecheck is more useful if they are specified:
```
def _create_predictions(country_reports: List[CountryReport]) -> Tuple[List[Dict], Dict[str, Dict]]:
```
Same for the missing return types in the functions above.